### DISCUSSION TITLE
Post Processing Query About Graphical Interface of Tidy3D

### DISCUSSION THREAD

**Discussion by khalid6955** [flexcompute/tidy3d#1186] [General]
Tidy3D is impressive cloud based computational approach for EM waves solver. I wanted to ask that, can we do post processing using the script command? In simple words, can we extract the field data from monitor using the Tidy3D script command once we complete the simulation utilizing the **Graphical User Interface** and do post-processing inside the Tidy3D.  For instance, we pick the example "Gradient metasurface reflector" and get the field data by placing a monitor in xy-plane. Can we inspect the farfield using the Tidy3D's builtin function"Near2Far" using script command?

**Comment by momchil-flex** [comment]
If you want to use scripting for post-processing, I think that it is really not that hard to just use the Tidy3D python frontend directly. You don't even need to install it if you don't want to you can _almost_ stay within our GUI. Namely, we provide a python environment for you with Tidy3D already installed. All you need to do is start it (see image)

![image](https://github.com/flexcompute/tidy3d/assets/92756559/145a12c0-e92e-4056-9151-bf8abd2bc536)

And then `sim_data = web.load(task_id)`, and do what you want with your data.

If you want to truly stay within the CAD GUI, I am not sure if we have this capability right now, or planning to have it. @xin-flex should know better.

By the way, specifically w.r.t. the near2far feature that you are asking about, note that the Near2Far post-processing command runs locally on CPU and could be slow for e.g. a large number of observation points/frequencies. It is much more efficient to use our field projection monitors to do the post-processing server side, hardware-accelerated. We are also planning to make this available as a standalone solver call (as opposed to only possible at the end of an FDTD simulation, currently), but this will take some time to be released.

**Comment by xin-flex** [comment]
Thanks @khalid6955 for your interest in our graphical user interface! Currently inside the GUI interface, our only scripting capability is to create structures/sources/monitors, and it does not support accessing data and post-processing yet (simple post-processing based on expressions is supported). Adding customized script-based post-processing in GUI is on our roadmap, but we don't have a clear date for its release yet. Please feel free to subscribe to our newsletter to get posted on our release updates:
<img width="272" alt="image" src="https://github.com/flexcompute/tidy3d/assets/102001380/052cfbc8-a772-45d5-ada5-3866db3e695f">
(at the bottom of this [webpage](https://www.flexcompute.com/tidy3d/solver/))

It might not be the most efficient approach for computation-intense tasks such as field projection calculation, and for this specific case, I think the methods mentioned by @momchil-flex (using field projection monitors, or in the future using the stand-alone field projection solver) are more efficient.

Does this solve your problem?

**Reply by khalid6955** [reply]
Thanks a lot for your promote feedback ( I really appricate it) and I have got my points. Glad to hear that the customized script-based post-processing in GUI is on your roadmap. This will really make the Tidy3D user friendly. In addtion, if you have any plan for workspcae (similar to Matlab) in GUI, it will grealty help the user to visulaize the data inise the Tidy3D, instead of using separete pyton based scripting.

**Comment by xin-flex** [comment]
Thanks @khalid6955 and good point! I think it (a "data visualizer") will be included naturally with script-based post-processing.
